python/cudf/cudf/core/buffer.py

```diff
@@ -72,23 +71,48 @@ def __init__(
                 raise TypeError("data must be Buffer, array-like or integer")
             self._init_from_array_like(np.asarray(data), owner)
 
+    @classmethod
+    def from_buffer(cls, buffer: Buffer, size: int = None, offset: int = 0):
+        """
+        Create a buffer from another buffer
+
+        Parameters
+        ----------
+        buffer : Buffer
+            The base buffer, which will also be set as the owner of
+            the memory allocation.
+        size : int, optional
+            Size of the memory allocation (default: `buffer.size`).
+        offset : int, optional
+            Start offset relative to `buffer.ptr`.
+        """
+
+        ret = cls()
+        ret._ptr = buffer._ptr + offset
+        ret.size = buffer.size if size is None else size
```

Comment:
I would consider requiring `size is not None` if `offset` is provided (or doing bounds checks).

`Buffer.from_buffer(buf, offset=32)` will create a buffer with the same size as `buf`, but its end points out of bounds for the original buffer. As a result, it's easier to misuse this API than a slice-based API like `buf[32:]`.